hoprnet/hoprnet
# time locking remote access

Closes #1986 

If an intruder tries to provide bad api token 5 or more times in a row, node remote access is time-locked for 5 minutes.

How to test:

```bash
for i in {1..5}
do
  websocat ws://127.0.0.1:3010/ -0 --header "Cookie:X-Auth-Token=bad_token
done
```

You should see something like

```
Locking websocket interface for 300000 ms due to 7 bad auth attempts
```

in the log.

For 5 minutes you shouldn't be able to use admin interface.

Same applied to REST API.
